src/websocket_service/mod.rs

```diff
@@ -37,10 +42,18 @@ pub struct RequestBody {
     pub params: String,
 }
 
+#[derive(Debug, Clone)]
+pub enum WebSocketClientState {
+    Connected,
+    Disconnected,
```

Comment:
non-blocking: I'm a little scared of this state being here. It doesn't look like this is ever set other than during startup, but it would be nice if there was an approach that avoided this extra state so it was always either Connected or Connecting.